src/internal_macros.rs

```diff
@@ -457,6 +457,170 @@ macro_rules! serde_string_impl {
     };
 }
 
+/// A combination of serde_struct_impl and serde_string_impl where string is
+/// used for human-readable serialization and struct is used for
+/// non-human-readable serialization.
+macro_rules! serde_struct_human_string_impl {
+    ($name:ident, $expecting:expr, $($fe:ident),*) => (
+        #[cfg(feature = "serde")]
+        impl<'de> $crate::serde::Deserialize<'de> for $name {
+            fn deserialize<D>(deserializer: D) -> Result<$name, D::Error>
+            where
+                D: $crate::serde::de::Deserializer<'de>,
+            {
+                if deserializer.is_human_readable() {
+                    use std::fmt::{self, Formatter};
```

Comment:
Perhaps I'm missing some nuance here, but why is this `std::...` while L507 uses `$crate::...`?

Reply:
Ah I just copied two existing macros in the if and else blocks. So I guess they weren't consistent. I don't know which is best. Probably `$crate::std`? Let me correct that.

Reply:
done in new commit